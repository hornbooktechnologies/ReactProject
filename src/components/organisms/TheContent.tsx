import React, { Suspense, useEffect, useState } from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'
import { Routes } from '../../route/routes'
import { localStorageUtils } from '../../utils/LocalStorageUtils'

const loading = (
   <div className="spinner">
      <div className="pt-3 text-center">
        <div uk-spinner="ratio: 3"></div>
      </div>
    </div>
  
)

const TheContent = () => {
  const [routes, setRoutes] = useState(Routes.getFixedRoutes())

  useEffect(() => {
    const userPref = localStorageUtils.getSelectedPlantUserPreferance()
    if (userPref) {
      let allRoutes = Routes.getAllRoutes(userPref)
      setRoutes(allRoutes)
    }
  }, [])

  return (
    <Suspense fallback={loading}>
      <Switch>
        {routes.map((route, idx) => {
          return (
            route.component && (
              <Route
                key={idx}
                path={route.path}
                exact={route.exact}
                name={route.name}
                render={(props) => <route.component {...props} />}
              />
            )
          )
        })}
        <Redirect from="/" to="/CurrentStatus" exact="true" />
        <Redirect from="//" to="/CurrentStatus" exact="true" />
      </Switch>
    </Suspense>
  )
}
export default React.memo(TheContent)
