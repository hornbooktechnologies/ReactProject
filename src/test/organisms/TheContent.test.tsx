import React from "react";
import renderer from "react-test-renderer";
import TheContent from "../../components/organisms/TheContent";
import { BrowserRouter } from "react-router-dom";

describe("Component: TheContent", () => {
  it("TheContent renders correctly", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <TheContent />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
