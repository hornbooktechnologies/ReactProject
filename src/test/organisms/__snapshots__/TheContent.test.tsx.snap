// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TheContent TheContent renders correctly 1`] = `
<div
  className="spinner"
>
  <div
    className="pt-3 text-center"
  >
    <div
      uk-spinner="ratio: 3"
    />
  </div>
</div>
`;
